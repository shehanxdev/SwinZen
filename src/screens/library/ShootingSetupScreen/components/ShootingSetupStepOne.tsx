import { View } from 'react-native';
import FastImage from 'react-native-fast-image';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, LandscapeImageUrl, TextAlignment, TextVariant } from '@sz/constants';

export function ShootingSetupStepOne() {
  return (
    <View>
      <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
        {` When setting up to shoot a video with a tripod (4-5ft height) or a buddy holding the phone, keep in mind these simple tips.\n`}
      </Text>
      <Text color={Color.Primary.Sz200} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
        {`Tip Number One:\n`}
      </Text>
      <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
        {`ALWAYS hold your phone in Landscape, NOT in Portrait. We understand that it feels more natural to hold the phone upright, but typically you cannot capture the full range of motion of your swing. Those frames that arenâ€™t caught when the ball and club are outside the camera shot are crucial to getting the best data feedback on your swing.\n`}
      </Text>
      <FastImage
        style={[tw`w-full border-8 border-Neutral-White`, { aspectRatio: 358 / 193 }]}
        source={{
          cache: FastImage.cacheControl.immutable,
          priority: FastImage.priority.high,
          uri: LandscapeImageUrl,
        }}
      />
    </View>
  );
}
