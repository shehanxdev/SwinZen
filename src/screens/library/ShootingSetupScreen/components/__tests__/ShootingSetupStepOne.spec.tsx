import { render } from '@testing-library/react-native';
import * as React from 'react';

import { ShootingSetupStepOne } from '../ShootingSetupStepOne';

describe('Shooting Setup Step One', () => {
  const getRenderedScreen = () => render(<ShootingSetupStepOne />);

  it(`should render Shooting Setup Step one correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
