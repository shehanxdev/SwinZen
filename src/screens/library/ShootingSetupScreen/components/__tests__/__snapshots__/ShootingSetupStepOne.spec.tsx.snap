// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shooting Setup Step One should render Shooting Setup Step one correctly 1`] = `
<View>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 18,
          "fontWeight": "normal",
          "textAlign": "auto",
        },
      ]
    }
  >
     When setting up to shoot a video with a tripod (4-5ft height) or a buddy holding the phone, keep in mind these simple tips.

  </Text>
  <Text
    style={
      [
        {
          "color": "#C0DBBF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "auto",
        },
      ]
    }
  >
    Tip Number One:

  </Text>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 18,
          "fontWeight": "normal",
          "textAlign": "auto",
        },
      ]
    }
  >
    ALWAYS hold your phone in Landscape, NOT in Portrait. We understand that it feels more natural to hold the phone upright, but typically you cannot capture the full range of motion of your swing. Those frames that arenâ€™t caught when the ball and club are outside the camera shot are crucial to getting the best data feedback on your swing.

  </Text>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderColor": "#FFFFFF",
          "borderWidth": 8,
          "height": 193,
          "width": "100%",
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "cache": "immutable",
          "priority": "high",
          "uri": "https://thumbnails.dev.swingzen.com/video-thumbs/landscape.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
